### DISCUSSION TITLE
source endpoint 242, not supported (zigate)

### DISCUSSION THREAD

**Discussion by Franckybel** [Koenkk/zigbee2mqtt#22759] [General]
Hello,

I have installed Home asistant and Z2MQTT using Zigate doogle with the latest firmware release Version 3.23

When I'm starting the service or when I'm doing any action I got the error message "source endpoint 242, not supported"

That's strange because it looks that the doogle is working, I have activated pairing and the first device was integrated on the network, but each action get always this error message

Thanks in advance for your support

**Logs in Home Assistant**
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/f7bbb10e-e2c5-41d5-b6cb-a782e617617e)

**Logs in Z2MQTT**
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/1112eec2-08ca-465b-a01c-eea6a782a8e7)

**Home Assistant Settings**
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/a72cc5af-b578-4cb8-86b3-c62e1b4a2772)

**First device integrated with success, basic but succeed **
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/27f033bc-b0f0-4057-a236-d2627c02a80f)

**Info Regarding Zigbee2MQTT**
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/670fb3fa-0f41-4177-b337-f16261434ef4)

**Info Regarding the doogle**
> Tout le matériel
Rechercher parmi le matériel
gpiochip0
gpiochip1
mice
ttyUSB0
/dev/serial/by-id/usb-Silicon_Labs_ZiGate_0001-if00-port0
Sous-système:
tty
Chemin d’accès au périphérique:
/dev/ttyUSB0
ID:
/dev/serial/by-id/usb-Silicon_Labs_ZiGate_0001-if00-port0
Attributs:
DEVLINKS: >-
  /dev/serial/by-id/usb-Silicon_Labs_ZiGate_0001-if00-port0
  /dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0-port0
  /dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usbv2-0:1.1:1.0-port0
DEVNAME: /dev/ttyUSB0
DEVPATH: >-
  /devices/platform/scb/fd500000.pcie/pci0000:00/0000:00:00.0/0000:01:00.0/usb1/1-1/1-1.1/1-1.1:1.0/ttyUSB0/tty/ttyUSB0
ID_BUS: usb
ID_MODEL: ZiGate
ID_MODEL_ENC: ZiGate
ID_MODEL_ID: ea60
ID_PATH: platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0
ID_PATH_TAG: platform-fd500000_pcie-pci-0000_01_00_0-usb-0_1_1_1_0
ID_PATH_WITH_USB_REVISION: platform-fd500000.pcie-pci-0000:01:00.0-usbv2-0:1.1:1.0
ID_REVISION: '0100'
ID_SERIAL: Silicon_Labs_ZiGate_0001
ID_SERIAL_SHORT: '0001'
ID_TYPE: generic
ID_USB_DRIVER: cp210x
ID_USB_INTERFACES: ':ff0000:'
ID_USB_INTERFACE_NUM: '00'
ID_USB_MODEL: ZiGate
ID_USB_MODEL_ENC: ZiGate
ID_USB_MODEL_ID: ea60
ID_USB_REVISION: '0100'
ID_USB_SERIAL: Silicon_Labs_ZiGate_0001
ID_USB_SERIAL_SHORT: '0001'
ID_USB_TYPE: generic
ID_USB_VENDOR: Silicon_Labs
ID_USB_VENDOR_ENC: Silicon\x20Labs
ID_USB_VENDOR_ID: 10c4
ID_VENDOR: Silicon_Labs
ID_VENDOR_ENC: Silicon\x20Labs
ID_VENDOR_ID: 10c4
MAJOR: '188'
MINOR: '0'
SUBSYSTEM: tty
TAGS: ':systemd:'
USEC_INITIALIZED: '34730809636'

**Comment by dough29** [comment]
Hello, I'm facing the same issue.

Here is my setup :

- ZiGate TTL-USB on port COM3 (Windows 11)
- zigbee2mqtt 1.38.0
- configuration.yaml

```yml
homeassistant: false
frontend: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://my_mqtt_server
serial:
  adapter: zigate
  port: \\.\COM3
```

Here is the log I get:

```
C:\Users\Damien\Desktop\zigbee2mqtt>npm start

> zigbee2mqtt@1.38.0 start
> node index.js

[2024-06-14 18:22:57] info:     z2m: Logging to console, file (filename: log.log)
[2024-06-14 18:22:57] info:     z2m: Starting Zigbee2MQTT version 1.38.0 (commit #fe048e6e)
[2024-06-14 18:22:57] info:     z2m: Starting zigbee-herdsman (0.49.2)
[2024-06-14 18:22:58] info:     zh:zigate: Connected to ZiGate adapter successfully.
[2024-06-14 18:22:58] error:    zh:zigate: [object Object]
[2024-06-14 18:22:58] info:     z2m: zigbee-herdsman started (resumed)
[2024-06-14 18:22:58] info:     z2m: Coordinator firmware version: '{"meta":{"maintrel":"2103","majorrel":"0","minorrel":"3","product":0,"revision":"2103","transportrev":0},"type":"zigate"}'
[2024-06-14 18:22:58] info:     z2m: Currently 0 devices are joined:
[2024-06-14 18:22:58] info:     z2m: Zigbee: disabling joining new devices.
[2024-06-14 18:22:58] error:    zh:zigate: source endpoint 242, not supported
[2024-06-14 18:22:58] info:     z2m:mqtt: Connecting to MQTT server at mqtt://my_mqtt_server
[2024-06-14 18:22:58] info:     z2m:mqtt: Connected to MQTT server
[2024-06-14 18:22:58] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'
[2024-06-14 18:22:58] info:     z2m: Started frontend on port 8080
[2024-06-14 18:22:58] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/config', payload '{"commit":"fe048e6e","coordinator":{"meta":{"maintrel":"2103","majorrel":"0","minorrel":"3","product":0,"revision":"2103","transportrev":0},"type":"zigate"},"log_level":"info","network":{"channel":11,"extendedPanID":"0x3d199f671906ff2a","panID":2836},"permit_join":false,"version":"1.38.0"}'
[2024-06-14 18:22:58] info:     z2m: Zigbee2MQTT started!
[2024-06-14 18:25:48] info:     z2m: Zigbee: allowing new devices to join.
[2024-06-14 18:25:48] error:    zh:zigate: source endpoint 242, not supported
[2024-06-14 18:25:48] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"g119q-1"}'
[2024-06-14 18:29:08] error:    zh:zigate: source endpoint 242, not supported
[2024-06-14 18:30:03] error:    zh:zigate: source endpoint 242, not supported
[2024-06-14 18:32:12] info:     z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'offline'
[2024-06-14 18:32:12] info:     z2m:mqtt: Disconnecting from MQTT server
[2024-06-14 18:32:12] info:     z2m: Stopping zigbee-herdsman...
[2024-06-14 18:32:12] error:    zh:zigate: source endpoint 242, not supported
Terminer le programme de commandes (O/N) ? [2024-06-14 18:32:12] info:  zh:zigate:driver: closing
[2024-06-14 18:32:12] info:     z2m: Stopped zigbee-herdsman
[2024-06-14 18:32:12] info:     z2m: Stopped Zigbee2MQTT
o

C:\Users\Damien\Desktop\zigbee2mqtt>
```
(I killed it with CTRL+C at 18:32:12)

Is this firmware related?

I read that ZiGate TTL-USB should be compatible with zigbee2mqtt but with which firmware version?

**Comment by AnthonyOdyssey** [comment]
Hello,
I've exactly the same issue using a ZIgate key with Z2M on Jeedom :

```
0000|>` zigbee2mqtt@1.38.0 start
0001|> node index.js
0002|[2024-06-21 08:16:52] e[31merrore[39m: 	z2m: Failed to load external converter file '/var/www/html/plugins/z2m/core/class/../config/converters/Danfoss/icon.js' (Cannot find module 'zigbee-herdsman-converters/lib/extend'
0003|Require stack:
0004|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/dist/util/utils.js
0005|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/dist/util/settings.js
0006|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/index.js)
0007|[2024-06-21 08:16:52] e[31merrore[39m: 	z2m: Probably there is a syntax error in the file or the external converter is not compatible with the current Zigbee2MQTT version
0008|[2024-06-21 08:16:52] e[31merrore[39m: 	z2m: Note that external converters are not meant for long term usage, it's meant for local testing after which a pull request should be created to add out-of-the-box support for the device
0009|[2024-06-21 08:16:54] e[31merrore[39m: 	zh:zigate: [object Object]
0010|[2024-06-21 08:16:54] e[31merrore[39m: 	zh:zigate: source endpoint 242, not supported
0011|[2024-06-21 08:22:09] e[31merrore[39m: 	zh:zigate: source endpoint 242, not supported
0012|> zigbee2mqtt@1.38.0 start
0013|> node index.js
0014|[2024-06-21 23:40:20] e[31merrore[39m: 	z2m: Failed to load external converter file '/var/www/html/plugins/z2m/core/class/../config/converters/Danfoss/icon.js' (Cannot find module 'zigbee-herdsman-converters/lib/extend'
0015|Require stack:
0016|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/dist/util/utils.js
0017|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/dist/util/settings.js
0018|- /var/www/html/plugins/z2m/resources/zigbee2mqtt/index.js)
0019|[2024-06-21 23:40:20] e[31merrore[39m: 	z2m: Probably there is a syntax error in the file or the external converter is not compatible with the current Zigbee2MQTT version
0020|[2024-06-21 23:40:20] e[31merrore[39m: 	z2m: Note that external converters are not meant for long term usage, it's meant for local testing after which a pull request should be created to add out-of-the-box support for the device
0021|[2024-06-21 23:40:24] e[31merrore[39m: 	zh:zigate:driver: Parsing error: Error: Message '8102' not found
0022|[2024-06-21 23:40:24] e[31merrore[39m: 	zh:zigate: [object Object]
0023|[2024-06-21 23:40:27] e[31merrore[39m: 	zh:zigate: source endpoint 242, not supported
```
Did you find any solution ?

**Comment by JPGNT** [comment]
Toujours le problème avec la V1.4.

**Comment by Arnaud13880** [comment]
Bonjour à tous,
J'ai le même problème que décrit ci dessus avec en plus le message suivant : "zh:zigate:driver: Parsing error: Error: Message '8008' not found"
Je pense que je vais finir par changer de dongle car ça devient vraiment fatigant d'avoir des périphériques qui se double et des déconnections intempestives.
Seul le fait de devoir réappairer tous mes périphériques me rebute un peu.

**Comment by dough29** [comment]
👋 
I switched to a SMLIGHT SLZB-07 dongle, working really fine, maybe the ZiGate dongle is now trash so we will not get any support...

**Comment by AnthonyOdyssey** [comment]
Je ne sais pas si cela peut aider, mais finalement pour moi cela fonctionne. Après quelques jours et quelques mises à jour des dépendances, la clé fonctionne et les périphériques se connectent.

**Reply by JPGNT** [reply]
En quoi une réponse aussi imprécise pourrait aider? Tu ne te rappelles même plus ce que tu as fait?

**Comment by AnthonyOdyssey** [comment]
Je crois avoir clairement écrit ce que j'ai fait. Je peux comprendre ta frustration sur le sujet, mais cela n'empêche pas la cordialité.
Je veux bien le dire autrement: j'ai mis à jour les dépendances plusieurs fois et j'ai attendu (au moins 24h). Je n'ai rien fait d'autre de notable.

**Reply by dough29** [reply]
Merci, j'ajoute mes petites questions : quel matériel est utilisé et quel firmware dans quelle version ?

**Reply by AnthonyOdyssey** [reply]
Hello,
J'utilise une ZIgate USB TTL de 2020, sur Jeedom Core beta 4.4.14. La version de Zigbee2MQTT est la 1.39.1.
Point important : je ne sais si sur les autres contrôleurs c'est identique, mais si je ne mets pas à jour Zigbee2MQTT à chaque nouvelle version, la connexion avec mes périphériques est perdue.
Si besoin d'autre chose, n'hésite pas.

**Reply by dough29** [reply]
Merci, et quel firmware sur le dongle ZiGate USL TTL ? 🤔

**Reply by AnthonyOdyssey** [reply]
J'arrive pas à retrouver l'information depuis que j'ai désinstaller le plugin Zigate. Voici que je peux trouver :
![image](https://github.com/user-attachments/assets/8eb28401-ac69-4337-be65-279b4331a0e4)

Si tu as une idée de l'endroit où je peux te trouver cette info...

**Comment by Franckybel** [comment]
J'avais ouvert ce sujet mais depuis cette période je suis passé sur une clé conbee II qui fonctionne parfaitement avec mes 32 devices.
Dommage car zigate est un projet Français mais il est impossible de trouver du support, donc passage à une clé 100% supportée par zm2 et pour seulement 25€

**Reply by JPGNT** [reply]
J'ai aussi une conbee en parallèle de ma Zigate  mais je trouve le truc moins ergonomique. Depuis la 1.40 de Z2M, le nombre de plantage sur ma conf est notoirement moindre, ainsi que les messages d'erruer? J'avais mis en place un programme de watchdog "maison" mais il n'a rien relancé depuis le 02/09 alors qu'avant c'était mini 2 plantages par jour. D'autre part, je pense avoir trouvé le device qui créait les messages "endpoint 242": c'était une prise de courant qui relève la conso. Elle  est vue comme une [Tuya TS011F] mais c'est une noname achetée sur Amazon.
